doc/sphinx/proofs/writing-proofs/rewriting.rst

```diff
@@ -762,58 +683,287 @@ the conversion in hypotheses :n:`{+ @ident}`.
+      | @ident
+
+   Generalizes the behavior of the :cmd:`Opaque` and :cmd:`Transparent`
+   commands. It is used to fine-tune the strategy for unfolding
+   constants, both at the tactic level and at the kernel level. This
+   command associates a :n:`@strategy_level` with the qualified names in the :n:`@reference`
+   sequence. Whenever two
+   expressions with two distinct head constants are compared (for
+   example, typechecking `f x` where `f : A -> B` and `x : C` will result in
+   converting `A` and `C`), the one
+   with lower level is expanded first. In case of a tie, the second one
+   (appearing in the cast type) is expanded.
+
+   This command accepts the :attr:`local` attribute, which limits its effect
+   to the current section or module, in which case the section and module
+   behavior is the same as :cmd:`Opaque` and :cmd:`Transparent` (without :attr:`global`).
+
+   Levels can be one of the following (higher to lower):
+
+    + ``opaque`` : level of opaque constants. They cannot be expanded by
+      tactics (behaves like +∞, see next item).
+    + :n:`@integer` : levels indexed by an integer. Level 0 corresponds to the
+      default behavior, which corresponds to transparent constants. This
+      level can also be referred to as ``transparent``. Negative levels
+      correspond to constants to be expanded before normal transparent
+      constants, while positive levels correspond to constants to be
+      expanded after normal transparent constants.
+    + ``expand`` : level of constants that should be expanded first (behaves
+      like −∞)
+    + ``transparent`` : Equivalent to level 0
+
+.. cmd:: Print Strategy @reference
+
+   This command prints the strategy currently associated with :n:`@reference`. It
+   fails if :n:`@reference` is not an unfoldable reference, that is, neither a
+   variable nor a constant.
+
+   .. exn:: The reference is not unfoldable.
+      :undocumented:
+
+.. cmd:: Print Strategies
+
+   Print all the currently non-transparent strategies.
+
+.. cmd:: Declare Reduction @ident := @red_expr
+
+   Declares a short name for the reduction expression :n:`@red_expr`, for
+   instance ``lazy beta delta [foo bar]``. This short name can then be used
+   in :n:`Eval @ident in` or ``eval`` constructs. This command
+   accepts the :attr:`local` attribute, which indicates that the reduction
+   will be discarded at the end of the
+   file or module. The name is not qualified. In
+   particular declaring the same name in several modules or in several
+   functor applications will be rejected if these declarations are not
+   local. The name :n:`@ident` cannot be used directly as an Ltac tactic, but
+   nothing prevents the user from also performing a
+   :n:`Ltac @ident := @red_expr`.
+
+   .. seealso:: :ref:`applyingconversionrules`
+
+.. todo PR
+   move rewriting.rst to equality.rst once the text is final,
+   perhaps in a different PR to keep the history somewhat understandable
```

Comment:
In order to reduce the delay before merging this PR, I suggest postponing the move of the "Equality" and "Equality and inductive sets" section to a subsequent PR (since I didn't even start reviewing those).

Reply:
I guess.  Making it a separate PR would make it easier to review that part.  OTOH I'd want to get those changes in soon as the next PR in the series.  Hopefully you'll have time to review those changes soon.

Reply:
Yes, I'm hopeful that it shouldn't take too long.